File: ngDailyTrainer/src/app/models/tdaquote.ts
import { last } from "rxjs";

export class TDAQuote {
  symbol:string;
  netChange:number;
  volatility:number;
  WkHigh52:number;
  WkLow52:number;
  peRatio:number;
  divAmount:number;
  divYield:number;
  divDate:string;
  lastPrice:number;
  description:string;
  totalVolume:number;
  openPrice:number;
  highPrice:number;
  lowPrice:number;
  closePrice:number;
  constructor(  symbol:string,
    netChange:number,
    volatility:number,
    WkHigh52:number,
    WkLow52:number,
    peRatio:number,
    divAmount:number,
    divYield:number,
    divDate:string,
    lastPrice:number,
    description:string,
    totalVolume:number, 
    openPrice:number,
    highPrice:number,
    lowPrice:number,
    closePrice:number){
      this.symbol = symbol;
      this.netChange = netChange;
      this.volatility = volatility;
      this.WkHigh52 = WkHigh52;
      this.WkLow52 = WkLow52;
      this.peRatio = peRatio;
      this.divAmount = divAmount;
      this.divYield = divYield;
      this.divDate = divDate;
      this.lastPrice = lastPrice;
      this.description = description;
      this.totalVolume = totalVolume;
      this.openPrice = openPrice;
      this.highPrice = highPrice;
      this.lowPrice = lowPrice;
      this.closePrice = closePrice;
    }



}

File: day-trader-monolith/DayTrainer/src/main/java/com/skilldistillery/daytrainer/tda/TDAQuote.java
package com.skilldistillery.daytrainer.tda;

import java.time.LocalDate;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@JsonIgnoreProperties(ignoreUnknown = true)
public class TDAQuote {
	
	private String symbol;
	
	private String description;
	
	private Double lastPrice;
	
	private Double openPrice;
	
	private Double closePrice;
	
	private Double lowPrice;
	
	private Double highPrice;
	
	private Long totalVolume;
	
	private Double netChange;
	
	private Double volatility;
	
	@JsonProperty("52WkHigh")
	Double weekHigh52;
	
	@JsonProperty("52WkLow")
	Double weekLow52;
	
	private Double peRatio;
	
	private Double divAmount;
	
	private String divDate;
	
	private double divYield;
	
}
